spec/support/controller_macros.rb

```diff
@@ -0,0 +1,11 @@
+# TODO:Try. See: https://github.com/plataformatec/devise/wiki/How-To:-Test-controllers-with-Rails-3-and-4-(and-RSpec)
+module ControllerMacros
+  def login_user
+    before(:each) do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+      user = FactoryGirl.create(:user)
+      # user.confirm! # or set a confirmed_at inside the factory. Only necessary if you are using the "confirmable" module
```

Comment:
Metrics/LineLength: Line is too long. [122/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)